### PR TITLE
generate add node info

### PR DESCRIPTION
**Pull request by jamesvan2019**

### CODE REVIEW

File: params/params.go [hunk 1]
```diff
@@ -287,14 +288,23 @@ type Params struct {
 	CoinbaseVersionConfig CoinbaseVersionConfig
 }
 
-type CoinbaseConfig struct {
+type CoinbaseHeightConfig struct {
```
Comment (dindinw) [inline params/params.go:291]: CoinbaseVersion

File: params/params.go [hunk 2]
```diff
@@ -287,14 +288,23 @@ type Params struct {
...
 	Height  int64
 	Version string
 }
 
-type CoinbaseVersionConfig []CoinbaseConfig
+type CoinbaseConfig struct {
+	Config   []CoinbaseHeightConfig
```
Comment (dindinw) [inline params/params.go:297]: versions []CoinbaseVersion

File: params/params.go [hunk 3]
```diff
@@ -287,14 +288,23 @@ type Params struct {
...
+	NodeInfo string
+	Enable   bool
+}
+
+type CoinbaseVersionConfig CoinbaseConfig
```
Comment (dindinw) [inline params/params.go:302]: naming

File: params/params.go [hunk 4]
```diff
@@ -287,14 +288,23 @@ type Params struct {
...
 
 func (cv *CoinbaseVersionConfig) Check(curHeight int64, coinbase []byte) bool {
```
Comment (dindinw) [inline params/params.go:304]: naming

File: params/params.go [hunk 5]
```diff
@@ -303,19 +313,30 @@ func (cv *CoinbaseVersionConfig) Check(curHeight int64, coinbase []byte) bool {
...
+		if cv.Config[i].Height > curHeight {
 			break
 		}
-		cc = (*cv)[i]
+		cc = cv.Config[i]
 	}
 	return cc.Version
 }
 
+func (cv *CoinbaseVersionConfig) String(curHeight int64) string {
```
Comment (dindinw) [inline params/params.go:329]: naming

File: p2p/service.go [hunk 6]
```diff
@@ -667,7 +667,7 @@ func NewService(cfg *config.Config, events *event.Feed, param *params.Params) (*
 	}
 
 	s.host = h
-
+	param.CoinbaseVersionConfig.SetNodeInfo(h.ID().String())
```
Comment (dindinw) [inline p2p/service.go:670]: the code is Intrusive,  you should depend on the p2p.api , instead of change the behavior internally.

**Review by dindinw — APPROVED** [review]

**Review by lochjin — APPROVED** [review]

**Pull request merged**